pandas/tests/series/test_api.py

```diff
@@ -117,9 +118,78 @@ def test_to_sparse_pass_name(self):
         result = self.ts.to_sparse()
         assert result.name == self.ts.name
 
+    def test_constructor_dict(self):
+        d = {'a': 0., 'b': 1., 'c': 2.}
+        result = self.series_klass(d)
+        expected = self.series_klass(d, index=sorted(d.keys()))
+        tm.assert_series_equal(result, expected)
```

Comment:
I think GitHub hid away my comment about this, but I think we should *not* interlace `assert_series_equal` and `assert_sp_series_equal`.  It reduces modularity, and that "assert_series_equal" is confusing for `SparseSeries` I find.

I propose that we do the following and define the following method:

~~~python
def assert_series_klass_equal(result, expected):
    klass_name = self.series_klass.__name__

    if klass_name == "Series":
        tm.assert_series_equal(result, expected)
    elif klass_name == "SparseSeries":
        tm.assert_sp_series_equal(result, expected)
    else:
        raise ValueError("Invalid 'series_klass' : {name}".format(name=klass_name))
~~~

That way you also don't need to modify `assert_series_equal`.